pkg/workflows/wasm/v2/runtime_wasip1.go

```diff
@@ -0,0 +1,134 @@
+package wasm
+
+import (
+	"errors"
+	"io"
+	"unsafe"
+
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/anypb"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
+	"github.com/smartcontractkit/chainlink-common/pkg/values/pb"
+	"github.com/smartcontractkit/chainlink-common/pkg/workflows/sdk/v2"
+	wpb "github.com/smartcontractkit/chainlink-common/pkg/workflows/wasm/v2/pb"
+)
+
+type runtimeBase struct {
+	execId string
+	config []byte
+}
+
+//go:wasmimport env call_capability
+func callCapability(req unsafe.Pointer, reqLen int32, id unsafe.Pointer) int64
+
+//go:wasmimport env await_capabilities
+func awaitCapabilities(awaitRequest unsafe.Pointer, awaitRequestLen int32, responseBuffer unsafe.Pointer, maxResponseLen int32) int64
+
+func (r *runtimeBase) CallCapability(request *wpb.CapabilityRequest) sdk.Promise[*wpb.CapabilityResponse] {
+	request.ExecutionId = r.execId
+	marshalled, err := proto.Marshal(request)
+	if err != nil {
+		return sdk.PromiseFromResult[*wpb.CapabilityResponse](nil, err)
+	}
+
+	id := make([]byte, sdk.IdLen)
+	marshalledPtr, marshalledLen, err := bufferToPointerLen(marshalled)
+	if err != nil {
+		return sdk.PromiseFromResult[*wpb.CapabilityResponse](nil, err)
+	}
+
+	result := callCapability(marshalledPtr, marshalledLen, unsafe.Pointer(&id[0]))
+	if result < 0 {
+		return sdk.PromiseFromResult[*wpb.CapabilityResponse](nil, errors.New("cannot find capability "+request.Id))
+	}
+
+	return sdk.NewBasicPromise(func() (*wpb.CapabilityResponse, error) {
+		awaitRequest := &wpb.AwaitCapabilitiesRequest{
+			ExecId: r.execId,
+			Ids:    []string{string(id)},
+		}
+		m, err := proto.Marshal(awaitRequest)
+		if err != nil {
+			return nil, err
+		}
+
+		mptr, mlen, err := bufferToPointerLen(m)
+		if err != nil {
+			return nil, err
+		}
+
+		// TODO make this configurable?
+		response := make([]byte, 2048)
```

Comment:
where does this length come from?

Reply:
It started as a PoC so I picked a number. I may have borrowed it from the current runtime, I don't remember.